internal/worker/worker.go

```diff
@@ -295,23 +353,39 @@ func (w *BaseWorker) gather(ctx context.Context, c string, numCmds int, ct CmdTy
 			w.logger.Debug("Error sending response to client", slog.String("workerID", w.id), slog.Any("error", err))
 			return err
 		}
-
-	case MultiShard:
+	} else if val.CmdType == MultiShard {
+		// Handle multi-shard commands
 		err := w.ioHandler.Write(ctx, val.composeResponse(evalResp...))
 		if err != nil {
 			w.logger.Debug("Error sending response to client", slog.String("workerID", w.id), slog.Any("error", err))
 			return err
 		}
-
-	default:
-		w.logger.Error("Unknown command type", slog.String("workerID", w.id))
+	} else {
+		w.logger.Error("Unknown command type", slog.String("workerID", w.id), slog.String("command", diceDBCmd.Cmd), slog.Any("evalResp", evalResp))
 		err := w.ioHandler.Write(ctx, diceerrors.ErrInternalServer)
 		if err != nil {
 			w.logger.Debug("Error sending response to client", slog.String("workerID", w.id), slog.Any("error", err))
 			return err
 		}
 	}
+	return nil
+}
+
+// handlePushResponse handles the response for push commands, i.e., responses for which are pushed from the server to the client.
+func (w *BaseWorker) handlePushResponse(ctx context.Context, cmdName, pushResponseKey string, evalResp []eval.EvalResponse) error {
```

Comment:
Do we need special handling for watch notification for handling response?
Can we use the the regular response flow instead?